templates/web-secrets.yaml

```diff
@@ -11,9 +11,6 @@ metadata:
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
-  host-key: {{ .Values.secrets.hostKey | b64enc | quote }}
```

Comment:
I thought `secrets.create` acutally meant "create the secrets with the hardcoded values". I overlooked the possibility of the user actually setting the secrets in values. You are right, this is not ok. How about:

- if `secrets.create` is true, `secrets.generate` is false and the user sets the secret values in the values.yaml we create the secret using the user provided values.
- if `secrets.create` is true, `secrets-generate` is false and there are not secrets set by the user, we produce an error (we don't provide no defaults)
- if `secrets.create` is true and `secrets-generate` is true we create the secret with values from the secrets-generator. We ignore user set values for secrets (if any).
- if `secrets.create` is false, we do nothing (we ignore `secrets.generate` and any secrets defined in values.yaml).

If feels complete to me but it will be a bit hard to explain in docs and maybe a bit hard for the user to grasp. The web-secret has so many keys in it that I wonder if it makes sense to expect from the users to create it themselves. With that in mind, maybe it makes sense to always create the secrets (even if empty) and populate the various keys with this logic:

- If `secrets.generate` is true and there is no value set in values.yaml try to generate it. If there is a value set in values.yaml use that (in a generate-if-needed concept)
- If `secrets.generate` is false don't try to generate missing values.

We can explain it in docs as "`secrets.generate` will try to generate SSH keys and session signing keys for you *if you don't set those manually in values.yaml*"

Reply:
I think this sounds good! I agree that it does add a good amount of complexity to secrets. Let's make sure the comments around these params clearly explain what they'll do. The values.yaml is already so large and hard for people to grok I feel...

> We can explain it in docs as "secrets.generate will try to generate SSH keys and session signing keys for you if you don't set those manually in values.yaml"
I like this, let's do it!